### DISCUSSION TITLE
GHelper v0.31 - UI Concept

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#91] [Polls]
Hello fellow community,

I have a feeling that some people struggle to find Optimized mode from Armoury (in G-Helper it's just a checkbox under GPU modes) or Auto refresh rate setting. Also it feels that too much checkboxes make UI cluttered (it has too many features already).

So I had in mind to do a small revamp of an interface

Thanks.

P.S. Recreated this post to add a Poll below, please VOTE.

## **[GHelper.zip](https://github.com/seerge/g-helper/releases/download/v0.31/GHelper.zip)**

![Screenshot 2023-03-17 224447](https://user-images.githubusercontent.com/5920850/226058845-177eece1-a8e7-4d7f-8351-716d8367647e.png)

<hr /><em>This discussion was created from the release <a href='https://github.com/seerge/g-helper/releases/tag/v0.30'>GHelper v0.30 - UI Concept</a>.</em>

**Comment by marcelomijas** [comment]
Now there is a lot of space wasted in the right side (it feels weird, unfinished). I think that for un-cluttering UI, it would be a better thing to have an UI option menu or some kind of thing to hide menus of features that you don't use or that some laptops don't support (i.e. GPU mode or laptop keyboard lights config on G14 2020 models).

![imagen](https://user-images.githubusercontent.com/62946623/226099592-b44eb089-1424-4b4a-b8ef-3069c45192f8.png)

Also, Windows 10 scaling doesn't do any justice to the icons.

P.S.: This is a personal opinion. Please, don't hurt me.

**Reply by TomBonnot** [reply]
I think that since it detects which version of laptop you have, he can adjusts the UI for this setup, since it is a try, I think it was programmed to re organise this. as for the upper buttons I like it, it feels smooth and feel like you are in peace using this soft, it fits the mood targeted by the soft.

**Reply by seerge** [reply]
> Now there is a lot of space wasted in the right side (it feels weird, unfinished). I think that for un-cluttering UI, it would be a better thing to have an UI option menu or some kind of thing to hide menus of features that you don't use or that some laptops don't support (i.e. GPU mode or laptop keyboard lights config on G14 2020 models).
> Also, Windows 10 scaling doesn't do any justice to the icons.

UI scaling is a good point. I have fixed most of complains - let me know if this version looks better. Thanks

[GHelper.zip](https://github.com/seerge/g-helper/files/11008690/GHelper.zip)

**Reply by marcelomijas** [reply]
Thank you! This version looks better. But personally, I think that "Fans + Power" button should be less tall or be like before, a fourth option in the "Performance Mode" section.

imagen

Also, another thing related to UI, in the "Fans + Power" menu, I think it would be great that when we touch a blue or red dot to adjust fan curves, a tag could tell us the coordinates of the dot in terms of RPM and temps (like Armoury Crate does). Something like this:

imagen

**Reply by seerge** [reply]
Problem that 4 buttons (with fans in it) breaks whole grid then. Or that's ok for you guys ?

![Screenshot 2023-03-18 155219](https://user-images.githubusercontent.com/5920850/226113409-104eb7b0-fe69-402f-993a-da834916785e.png)

**Reply by marcelomijas** [reply]
Personally, I don't have a problem with the grid breaking as long as there are no big wasted spaces... Others might disagree.

**Reply by seerge** [reply]
> Personally, I don't have a problem with the grid breaking as long as there are no big wasted spaces... Others might disagree.

Ok, I have added a tooltips for chart btw 
[GHelper.zip](https://github.com/seerge/g-helper/files/11008919/GHelper.zip)

**Reply by marcelomijas** [reply]
Thanks! This is next level man! :D

**Comment by AreyUA** [comment]
Is it possible to add GPU overclocking for the 2021 model?

**Reply by TomBonnot** [reply]
We did not succeed into reaching the right endpoint for this. if you want to try and help, you could go into the #38 channel

**Reply by seerge** [reply]
> Is it possible to add GPU overclocking for the 2021 model?

I'm not a big fan of blindly adding something I can't test at all (as owner of 2022 model only). But https://github.com/JamesCJ60/Universal-x86-Tuning-Utility/releases seem to be able to adjust GPU clocks. You can try that :)

**Comment by Protector980** [comment]
I think the new UI is great, thank you for this great program. I know it's not an option in armoury crate but is it possible to add the ability to undervolt the cpu like uxtu does? It would be really nice to have everything in one application.

**Reply by seerge** [reply]
That's absolutely separate topic. And - no, I don't plan to add any hardcore tuning here (or lots of people will start blaming app for breaking their laptops :D) . Current aim - is just do what armory does :)

**Comment by hkaancaliskan** [comment]
I have GA401QC and I have only 60hz display. 120HZ+OD option does nothing for me. Laptop screen section can be hidden for 60hz displays.

**Reply by seerge** [reply]
I'm looking at specs and it says 120Hz is supported https://tweakers.net/pricewatch/1793892/asus-rog-zephyrus-g14-ga401qc-k2175w/specificaties/ are you sure that it's not drivers messing up ?

**Reply by hkaancaliskan** [reply]
there are 2 types of display, 120hz and 60hz
mine is this one https://kelaptop.com/en/asus-rog-zephyrus-g14-ga401qc-bm205w-90nr05t3-m004m0

**Reply by seerge** [reply]
Oh, ok :) Anyway, current "max refresh rate" detection is a bit tricky, and if I will start hiding that section it may break behavior for some cases even when screen supports 120hz :( so for now it will stay.

**Comment by TomBonnot** [comment]
I was thinking of something. 
but the Optimized mode is about low consummation on battery and max perfs on plugged. 
The Auto button is more or less the same but for the screen. Wouldn’t it make more sense to have the Auto button on the right side, aligned with the Optimized one ? 
This way we have all low consummations mods on the left side !

**Comment by LJGJM** [comment]
> Hello fellow community,
> 
> I have a feeling that some people struggle to find Optimized mode from Armoury (in G-Helper it's just a checkbox under GPU modes) or Auto refresh rate setting. Also it feels that too much checkboxes make UI cluttered (it has too many features already).
> 
> So I had in mind to do a small revamp of an interface
> 
> Thanks.
> 
> P.S. Recreated this post to add a Poll below, please VOTE.
> 
> ## **[GHelper.zip](https://github.com/seerge/g-helper/releases/download/v0.31/GHelper.zip)**
> ![Screenshot 2023-03-17 224447](https://user-images.githubusercontent.com/5920850/226058845-177eece1-a8e7-4d7f-8351-716d8367647e.png)
> 
> _This discussion was created from the release [GHelper v0.30 - UI Concept](https://github.com/seerge/g-helper/releases/tag/v0.30)._

Is there a way to undervolt this CPU? (amd 4600h)

**Reply by lilzoe5** [reply]
Looks nice, only thing I would change is the "fans+power" box width and maybe to a different locations like a smaller button somewhere. Also, a dark mode would be nice too like how Windows control center has it
![image](https://user-images.githubusercontent.com/47486913/226152916-dcb6a2c8-a8a7-45d4-b88b-4c4c4c39fbeb.png)

**Reply by koezgen** [reply]
That aside, wouldn't it be insane if it was possible to just add these functions directly on the windows control center? like below all the defaults?

**Reply by seerge** [reply]
I don't think that's doable, at least definitely not on that custom level as g-helper would need :)
